gflownet/envs/crystals/spacegroup.py

```diff
@@ -656,11 +673,92 @@ def build_n_atoms_compatibility_dict(n_atoms: List[int], space_groups: List[int]
         assert all([sg > 0 and sg <= 230 for sg in space_groups])
         return {sg: Group(sg).check_compatible(n_atoms)[0] for sg in space_groups}
 
+    def _restrict_space_groups(self, sg_subset: Optional[Iterable] = None):
+        """
+        Updates the dictionaries:
+            - self.space_groups
+            - self.crystal_lattice_systems
+            - self.point_symmetries
+        by eliminating the space groups that are not in the subset sg_subset passed as
+        an argument.
+        """
+        if sg_subset is None:
+            return
```

Comment:
So once a SpaceGroup has been instantiated, we can't go back and change the space group restrictions?
Just want to clarify because changing atoms after `__init__` seems supported but not the sg_subset. We should probably document somewhere what can and can't be modified after init to avoid weird errors.

Or, instead, this method could set `self.space_groups`, `self.crystal_lattice_systems` and `self.point_symmetries` (and not merely restrict them) based on the provided `sg_subset`. I think that it's basically adding 3 lines at the start of this method : 
```
self.crystal_lattice_systems = _get_crystal_lattice_systems()
self.point_symmetries = _get_point_symmetries()
self.space_groups = _get_space_groups()
```
and it would protect us against headaches down the line if someone tries to change the valid space groups for a Spacegroup env that already exists.

Reply:
I agree @carriepl-mila